### PR TITLE
Menubar Designed

### PR DESCRIPTION
**Pull request by MarcoPolo**
## Preview
screen shot 2015-12-15 at 4 01 32 pm

(The debug button won't show up in the prod version)

## New things

* Menubar's popup will auto update on height change.
* ipcMain listens for 'closeMenubar'

Since the menubar popup is just a react component we can even put it in the components sheet, so I did :)

@keybase/react-hackers & @cecileboucheron

### CODE REVIEW

**Comment by cecileboucheron** [comment]
Looking great!! 
Any chance we can push down the folder icons a little bit? I think they're a bit too high. I would say 3px down.Apart from this, it's great for me.

File: react-native/react/images/folders/fa-folder-private-empty.svg [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (chrisnojima) [inline react-native/react/images/folders/fa-folder-private-empty.svg]: @cecileboucheron is this from font-awesome or is this a our own? If it is from font-awesome @MarcoPolo we already have that so we shouldn't need to have our own copy. If it is our own we likely shouldn't name this fa- as that's for font-awesome stuff

File: react-native/react/menubar/index.js [hunk 2]
```diff
@@ -1,20 +1,43 @@
...
   props: MenubarProps;
   render () {
-    return <Render {...this.props}/>
+    const openingMessage = this.props.username ? 'Be sure to drink your Ovaltine.' : "Looks like you aren't logged in. Try running `keybase login`"
+
+    const closeMenubar = () => ipcRenderer.send('closeMenubar')
+
+    // TODO change this to /keybase when all our mount points are `/keybase/...`
+    // TODO Support linux
+    const kbfsPath = `${process.env.HOME}/keybase`
```
Comment (chrisnojima) [inline react-native/react/menubar/index.js]: so this moved to /keybase already, but ultimately we should get this from the daemon. its fine for this to be hardcoded now but likely we should move it out

Comment (chrisnojima) [inline react-native/react/menubar/index.js]: actually let's at least move this to constants

File: react-native/react/menubar/index.render.desktop.js [hunk 3]
```diff
@@ -1,50 +1,78 @@
...
   render (): ReactElement {
+    const {openKBFS, openKBFSPublic, openKBFSPrivate, showMain, showHelp, quit, openingButtonInfo} = this.props
+
     return (
-      <div>
-        <Header/>
-        <OpeningMessage/>
-        <FolderList/>
-        <Footer debug={this.props.debug || false}/>
+      <div style={{backgroundColor: '#f6f6f4', width: 320}}>
```
Comment (chrisnojima) [inline react-native/react/menubar/index.render.desktop.js]: we shoudl discuss the styling. we shouldn't have hardcoded colors etc in here. we have colors injected into the commonStyles

File: react-native/react/menubar/index.render.desktop.js [hunk 4]
```diff
@@ -1,50 +1,78 @@
...
   render () {
+    const {buttonInfo} = this.props
     return (
-      <div>
-        <div> Success! Some cool message goes here on two lines. </div>
+      <div style={{backgroundColor: '#00bff0', display: 'flex', flexDirection: 'column', alignItems: 'center'}}>
+        <div style={{textAlign: 'center', color: '#ffffff', marginTop: 18, marginBottom: 11, marginRight: 32, marginLeft: 32}}>{this.props.message}</div>
         <div>
-          <FlatButton style={commonStyles.primaryButton} label='getStarted'/>
+          {buttonInfo && <FlatButton style={{...commonStyles.primaryButton, ...NotoSans, fontSize: 17, marginBottom: 22}} onClick={buttonInfo.onClick} label={buttonInfo.text}/>}
```
Comment (chrisnojima) [inline react-native/react/menubar/index.render.desktop.js]: we already have a common style for primary buttons

File: react-native/react/menubar/index.render.desktop.js [hunk 5]
```diff
@@ -52,29 +80,86 @@ class OpeningMessage extends Component {
...
+      <div style={{display: 'flex', flexDirection: 'row', justifyContent: 'space-between', paddingLeft: 11, paddingRight: 11, paddingTop: 8, paddingBottom: 8}}>
+        <div style={{...Clickable, ...TransparentBlack, ...NotoSans, fontSize: 13, lineHeight: '17px'}} onClick={this.props.showHelp}>Help</div>
+        {this.props.debug && <div style={{...Clickable, ...TransparentBlack, ...NotoSans, fontSize: 13, lineHeight: '17px'}} onClick={this.props.showMain}>Debug</div>}
+        <div style={{...Clickable, ...TransparentBlack, ...NotoSans, fontSize: 13, lineHeight: '17px'}} onClick={this.props.quit}>Quit</div>
       </div>
     )
   }
 }
+
+const Clickable = {
```
Comment (chrisnojima) [inline react-native/react/menubar/index.render.desktop.js]: we already have a common style for clickable

File: react-native/react/menubar/index.render.desktop.js [hunk 6]
```diff
@@ -52,29 +80,86 @@ class OpeningMessage extends Component {
...
+  cursor: 'pointer'
+}
+
+const NotoSans = {
```
Comment (chrisnojima) [inline react-native/react/menubar/index.render.desktop.js]: we aleady have a style for font stuff

File: react-native/react/menubar/index.render.desktop.js [hunk 7]
```diff
@@ -52,29 +80,86 @@ class OpeningMessage extends Component {
...
+  fontFamily: 'Noto Sans'
+}
+
+const TransparentBlack = {
```
Comment (chrisnojima) [inline react-native/react/menubar/index.render.desktop.js]: we have common styles for colors

File: react-native/react/menubar/index.render.desktop.js (hunk 4 above)
Comment (MarcoPolo) [inline react-native/react/menubar/index.render.desktop.js]: This uses that, but overwrites some values to match the design.

File: react-native/react/images/folders/fa-folder-private-empty.svg (hunk 1 above)
Comment (cecileboucheron) [inline react-native/react/images/folders/fa-folder-private-empty.svg]: No these are SVG icons that I created (from font awesome ones).

Comment (chrisnojima) [inline react-native/react/images/folders/fa-folder-private-empty.svg]: @MarcoPolo let's rename them

File: react-native/react/menubar/index.render.desktop.js (hunk 4 above)
Comment (chrisnojima) [inline react-native/react/menubar/index.render.desktop.js]: font size is fine (but maybe we should talk to design) but i think noto sans should get pickedup. i think its applied at a top level in style.css. maybe thats not being loaded in the remote components?

File: react-native/react/images/folders/fa-folder-private-empty.svg (hunk 1 above)
Comment (MarcoPolo) [inline react-native/react/images/folders/fa-folder-private-empty.svg]: I think this private-empty is the same as the font-awesome one, but the other svgs are different. Having this one as an svg makes the positioning consistent across the images.

Comment (chrisnojima) [inline react-native/react/images/folders/fa-folder-private-empty.svg]: i'm fine if they're copies, but let's not name them fa- cause they're not font-awesome things and that's confusing

**Comment by MarcoPolo** [comment]
@chrisnojima changed all the style stuff, tal please?

**Comment by chrisnojima** [comment]
:+1:

**Pull request merged**